modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -684,13 +698,38 @@ private void _addObjectDefinitionResources(
 				).build();
 
 			for (ObjectRelationship objectRelationship : objectRelationships) {
+				String objectRelationshipName = objectRelationship.getName();
+
+				com.liferay.object.model.ObjectRelationship
+					oldObjectRelationship =
+						_objectRelationshipLocalService.
+							fetchObjectRelationshipByObjectDefinitionId(
+								objectDefinitionId, objectRelationshipName);
+
+				if (!deleteRelationships.isEmpty() &&
+					!deleteRelationships.contains(objectRelationshipName)) {
```

Comment:
@mateussgn why do we need to check if the deleteRelationships is empty? if we need, why you don't do the same check for putRelationships?

Reply:
@mateussgn why you're negating the condition deleteRelationships.contains(objectRelationshipName)? I mean if this condition is true we should proceed and delete the relationship, right?

Reply:
1. By using `deleteRelationships.isEmpty()` we are ensuring that we have relationships to delete.
Otherwise, with an empty list the next method call (the `deleteRelationships.contains(objectRelationshipName)`) will aways return `false` and all relationships would be deleted.
What I can do to make the code more readable is delete the object relationships from the `deleteRelationships` list in a loop apart. Let's say something like this:
```java
for (String deleteRelationship : deleteRelationships) {
       com.liferay.object.model.ObjectRelationship
		oldObjectRelationship =
			_objectRelationshipLocalService.
				fetchObjectRelationshipByObjectDefinitionId(
					objectDefinitionId, oldObjectRelationshipsName);

	_objectRelationshipLocalService.deleteObjectRelationship(
			oldObjectRelationship.getObjectRelationshipId()
}
```
And then I should call the loop with the put and post methods.